### ISSUE TITLE
MTR

### ISSUE THREAD

**Issue by YibingZeng** [guoweilong/cgmaptools#37]
Hi Dr. Guo,

When I tried to do the MTR analysis, I find out result as followed:
`1       21800   21991   NA      0       NA      0`

But when I trace back to the CGmap.gz file, I can clearly see that there is methyaltion there:
1	C	21823	CG	CG	1.0	1	1
1	G	21824	CG	CG	1.0	2	2
1	C	21845	CG	CG	1.0	1	1
1	G	21846	CG	CG	1.0	2	2
1	C	21849	CG	CG	1.0	2	2
1	G	21850	CG	CG	0.5	1	2
1	C	21851	CG	CG	1.0	2	2
1	G	21852	CG	CG	1.0	2	2
1	C	21853	CG	CG	1.0	2	2
1	G	21854	CG	CG	0.0	0	2
1	C	21882	CG	CG	1.0	1	1
1	G	21883	CG	CG	1.0	1	1
1	C	21897	CG	CG	1.0	2	2
1	C	21912	CG	CG	1.0	2	2
1	C	21914	CG	CG	1.0	2	2
1	C	21926	CG	CG	1.0	2	2
1	C	21939	CG	CG	1.0	2	2
1	C	21953	CG	CG	1.0	2	2
1	C	21975	CG	CG	1.0	4	4

The command I used for the analysis is:
`cgmaptools mtr -i rep3CHG.gz -r ${MITE} -o B73_rep3_CHG_MITE.mtr`

Would you please tell me how can I fix this problem?

Yibing

**Comment by guoweilong** [comment]
Hi Yibing,

**Comment by guoweilong** [comment]
Hi Yibing,

Are you working on correct files?
Your file name is "CHG", but all the sites you pasted here are in CG context.

Best,
Weilong Guo

**Comment by YibingZeng** [comment]
Sorry for that, the command I use is:
`cgmaptools mtr -i rep1CG.gz -r ${MITE} -o B73_rep1_CG_MITE.mtr`

**Comment by YibingZeng** [comment]
#19 Provide two methods to fix the problem, one is to change python3 to python2, another one is to sort the file. I think I will try them. If you have better solution, please let me know as well. 
Hope you do well.
Yibing

**Comment by guoweilong** [comment]
Yes, python version and sorting seems be required. Wish to read your feedback.

**Comment by YibingZeng** [comment]
It works. But if the regions are overlapping, only the first region would be calculated. The problem of there is coverage but could not be called was solved!

**Comment by guoweilong** [comment]
Em, yes. Current version is for sorted non-overlapping regions. Maybe we'll figure out how to applied it to regions with overlap.
